### PR TITLE
add support for exporting to json and csv in workspace chats

### LINKED ISSUE #596: [FEAT]: Filtering the conversation and downloading JSON after filter
### What would you like to see?

Screenshot 2024-01-14 123244
there should be a provision to filter the conversations based on the User and the Workspace. The filtered JSON should be allowed to be downloaded.

### PR DESCRIPTION
**Pull request by shatfield4**
### Pull Request Type

- [x] ✨ feat
- [ ] 🐛 fix
- [ ] ♻️ refactor
- [ ] 💄 style
- [ ] 🔨 chore
- [ ] 📝 docs

### Relevant Issues

resolves #596 

### What is in this change?

Describe the changes in this PR that are impactful to the repo.

- Add options to export workspace chats as JSON and CSV

### Additional Information

Add any other context about the Pull Request here that was not captured above.

### Developer Validations

- [x] I ran `yarn lint` from the root of the repo & committed changes
- [x] Relevant documentation has been updated
- [x] I have tested my code functionality
- [x] Docker build succeeds locally

### CODE REVIEW

File: frontend/src/models/system.js [hunk 1]
```diff
@@ -423,8 +423,9 @@ const System = {
         return { success: false, error: e.message };
       });
   },
-  exportChats: async () => {
-    return await fetch(`${API_BASE}/system/export-chats`, {
+  exportChats: async (type = "csv") => {
+    const url = `${API_BASE}/system/export-chats?type=${type}`;
+    return await fetch(url, {
```
Comment (timothycarambat) [inline frontend/src/models/system.js:428]: Use search params constructor to encode + safely encode URL options
```
const url = new URL(`${fullApiUrl()}/system/export-chats`);
 url.searchParams.append("type", encodeURIComponent(type));
 ```

File: frontend/src/pages/GeneralSettings/Chats/index.jsx [hunk 2]
```diff
@@ -7,29 +7,60 @@ import useQuery from "@/hooks/useQuery";
...
-      link.download = "chats.jsonl";
+      link.download = `chats.${fileExtension}`;
       document.body.appendChild(link);
       link.click();
       window.URL.revokeObjectURL(link.href);
       document.body.removeChild(link);
       showToast(
-        "Chats exported successfully. Note: Must have at least 10 chats to be valid for OpenAI fine tuning.",
+        `Chats exported successfully as ${fileExtension.toUpperCase()}. Note: Must have at least 10 chats to be valid for OpenAI fine tuning.`,
         "success"
```
Comment (timothycarambat) [inline frontend/src/pages/GeneralSettings/Chats/index.jsx:36]: Lets drop the `10 chats` part about openAI. Lets just say chats were exported

File: frontend/src/pages/GeneralSettings/Chats/index.jsx [hunk 3]
```diff
@@ -7,29 +7,60 @@ import useQuery from "@/hooks/useQuery";
...
+        menuRef.current &&
+        !menuRef.current.contains(event.target) &&
+        !openMenuButton.current.contains(event.target)
+      ) {
+        setShowMenu(false);
+      }
+    }
+
+    document.addEventListener("mousedown", handleClickOutside);
+    return () => {
```
Comment (timothycarambat) [inline frontend/src/pages/GeneralSettings/Chats/index.jsx:59]: Does this give you an error in console? It does for me sometimes

File: server/endpoints/system.js [hunk 4]
```diff
@@ -878,13 +880,27 @@ function systemEndpoints(app) {
...
+            output = await convertToCSV(workspaceChatsMap);
+            break;
+          }
+          // JSONL default
+          default: {
+            response.setHeader("Content-Type", "application/jsonl");
+            output = await convertToJSONL(workspaceChatsMap);
+            break;
+          }
+        }
```
Comment (timothycarambat) [inline server/endpoints/system.js:901]: :+1

File: server/endpoints/system.js [hunk 5]
```diff
@@ -826,8 +827,9 @@ function systemEndpoints(app) {
   app.get(
     "/system/export-chats",
     [validatedRequest, flexUserRoleValid],
-    async (_request, response) => {
+    async (request, response) => {
       try {
+        const { type = "csv" } = request.query;
```
Comment (timothycarambat) [inline server/endpoints/system.js:832]: If JSONL is the default case, it should be here too

**Review by timothycarambat — CHANGES_REQUESTED** [review]
See comments

File: frontend/src/pages/GeneralSettings/Chats/index.jsx (hunk 3 above)
Comment (shatfield4) [reply frontend/src/pages/GeneralSettings/Chats/index.jsx:59]: Hmm I don't get any errors in console with this for any case.

**Review by timothycarambat — APPROVED** [review]

**Pull request merged**